iceoryx_posh/test/moduletests/test_popo_base_publisher.cpp

```diff
@@ -0,0 +1,240 @@
+    }
+    port_t& getMockedPort()
+    {
+        return iox::popo::BasePublisher<T, port_t>::m_port;
+    }
+};
+
+using TestBasePublisher = StubbedBasePublisher<DummyData, MockPublisherPortUser>;
+
+class BasePublisherTest : public Test
+{
+  public:
+    BasePublisherTest()
+    {
+    }
+
+    void SetUp()
+    {
+    }
+
+    void TearDown()
+    {
+    }
+
+  protected:
+    TestBasePublisher sut{{"", "", ""}};
+};
+
+TEST_F(BasePublisherTest, LoanForwardsAllocationErrorsToCaller)
+{
+    // ===== Setup ===== //
+    ON_CALL(sut.getMockedPort(), tryAllocateChunk)
+        .WillByDefault(Return(
+            ByMove(iox::cxx::error<iox::popo::AllocationError>(iox::popo::AllocationError::RUNNING_OUT_OF_CHUNKS))));
+    // ===== Test ===== //
+    auto result = sut.loan(sizeof(DummyData));
+    // ===== Verify ===== //
+    EXPECT_EQ(true, result.has_error());
+    EXPECT_EQ(iox::popo::AllocationError::RUNNING_OUT_OF_CHUNKS, result.get_error());
+    // ===== Cleanup ===== //
+}
+
+TEST_F(BasePublisherTest, LoanReturnsAllocatedTypedSampleOnSuccess)
+{
+    // ===== Setup ===== //
+    auto chunk =
+        reinterpret_cast<iox::mepoo::ChunkHeader*>(iox::cxx::alignedAlloc(32, sizeof(iox::mepoo::ChunkHeader)));
+    ON_CALL(sut.getMockedPort(), tryAllocateChunk)
+        .WillByDefault(Return(ByMove(iox::cxx::success<iox::mepoo::ChunkHeader*>(chunk))));
+    // ===== Test ===== //
+    auto result = sut.loan(sizeof(DummyData));
+    // ===== Verify ===== //
+    // The memory location of the sample should be the same as the chunk payload.
+    EXPECT_EQ(chunk->payload(), result.get_value().get());
+    // ===== Cleanup ===== //
+    iox::cxx::alignedFree(chunk);
+}
+
+TEST_F(BasePublisherTest, LoanedSamplesContainPointerToChunkHeader)
+{
+    // ===== Setup ===== //
+    auto chunk =
+        reinterpret_cast<iox::mepoo::ChunkHeader*>(iox::cxx::alignedAlloc(32, sizeof(iox::mepoo::ChunkHeader)));
+    ON_CALL(sut.getMockedPort(), tryAllocateChunk)
+        .WillByDefault(Return(ByMove(iox::cxx::success<iox::mepoo::ChunkHeader*>(chunk))));
+    // ===== Test ===== //
+    auto result = sut.loan(sizeof(DummyData));
+    // ===== Verify ===== //
+    EXPECT_EQ(chunk, result.get_value().getHeader());
+    // ===== Cleanup ===== //
+    iox::cxx::alignedFree(chunk);
+}
+
+TEST_F(BasePublisherTest, LoanedSamplesAreAutomaticallyReleasedWhenOutOfScope)
+{
+    // ===== Setup ===== //
+    auto chunk =
+        reinterpret_cast<iox::mepoo::ChunkHeader*>(iox::cxx::alignedAlloc(32, sizeof(iox::mepoo::ChunkHeader)));
+
+    ON_CALL(sut.getMockedPort(), tryAllocateChunk)
+        .WillByDefault(Return(ByMove(iox::cxx::success<iox::mepoo::ChunkHeader*>(chunk))));
+
+#if defined(__APPLE__)
+    // Required because for some reason, only on Mac, free chunk is called repeatedly by google test.
+    // Since this problem is not reproducable in Linux, it is assumed that it is a false positive.
+    EXPECT_CALL(sut.getMockedPort(), freeChunk(chunk));
```

Comment:
Please write:
EXPECT_CALL(sut.getMockedPort(), freeChunk(chunk)).Times(AtLeast(1)); 

To mitigate the problem with MacOS.